docs/understanding-airbyte/airbyte-protocol.md

```diff
@@ -802,8 +807,47 @@ AirbyteErrorTraceMessage:
       enum:
         - system_error
         - config_error
+AirbyteEstimateTraceMessage:
+  type: object
+  additionalProperties: true
+  required:
+    - name
+    - type
+  properties:
+    name:
+      description: The name of the stream
+      type: string
+    type:
+      description: The type of estimate
+      type: string
+      enum:
+        - STREAM
+        - SYNC
+    namespace:
+      description: The namespace of the stream
+      type: string
+    row_estimate:
+      description: The estimated number of rows to be emitted by this sync for this stream
+      type: integer
+    byte_estimate:
+      description: The estimated number of bytes to be emitted by this sync for this stream
+      type: integer
 ```
 
+### AirbyteErrorTraceMessage
+
+Error Trace Messages are used when a sync is about to fail and the connector can provide meaningful information to the orhcestrator or user about what to do next.
+
+Of note, an `internal_message` might be an exception code, but an `external_message` is meant to be user-facing, e.g. "Your API Key is invalid".
+
+Syncs can fail for multiple reasons, and therefore multiple `AirbyteErrorTraceMessage` can be sent from a connector.
+
+### AirbyteEstimateTraceMessage
+
+Estimate Trace Messages are used by connectors to inform the orchestrator about how much data they expect to move within the sync. This ise useful to present the user with estimates of the time remaining in the sync, or percentage complete. An example of this would be for every stream about to be synced from a databse to provde a `COUNT (*) from {table_name} where updated_at > {state}` to provide an estimate of the rows to be sent in this sync.
+
+`AirbyteEstimateTraceMessage` should be emitted early in th sync to provide an early estimate of the sync's duration. Multiple `AirbyteEstimateTraceMessage`s can be sent for the same stream, and an updated estimate will replace the previous value.
```

Comment:
```suggestion
`AirbyteEstimateTraceMessage` should be emitted early in the sync to provide an early estimate of the sync's duration. Multiple `AirbyteEstimateTraceMessage`s can be sent for the same stream, and an updated estimate will replace the previous value.
```

Reply:
to Charles' comment - I think it's worth specifying here how the `STREAM` and `SYNC` enums are meant to be used.